src/lib/backoff.ts

```diff
@@ -0,0 +1,19 @@
+import { log } from '../lib/supervisor-console';
+
+/**
+ * Perform exponential backoff on the function, increasing the attempts
+ * counter on each call
+ *
+ * If attempts is 0 then, it will delay for min(minDelay, maxDelay)
+ */
+export const backoff = (
+	retry: (attempts: number) => void,
+	attempts = 0,
+	maxDelay: number,
+	minDelay: number,
+	name: string,
+) => {
+	const delay = Math.min(2 ** attempts * minDelay, maxDelay);
+	log.info(`${name}: Retrying in ${delay / 1000} seconds`);
```

Comment:
I'm not sure is that important right now, but I think it would be better if we passed something like an `onError` callback to this function, so this module would be independent of the `supervisor-console` module.

I we did that we could call this function as

```
backoff(
  reportCurrentState,
  stateReportErrors++,
  maxDelay,
  MINIMUM_BACKOFF_DELAY,
 (delay). => `Retrying in ${delay/1000} seconds`)
```

which would make it a bit more generic

Reply:
I agree